packages/shared/src/observe-slot-text.ts

```diff
@@ -15,7 +15,7 @@ const observedSlotElement = Symbol('observedSlotElement');
 const slotElementObserver = Symbol('slotElementObserver');
 const startObserving = Symbol('startObserving');
 
-type Constructor<T = object> = {
+type Constructor<T = Record<string, unknown>> = {
```

Comment:
curious about this change, is this a new change in typescript? should we be typing the record to support keys aside from string (i.e. symbol/numbers)?

Reply:
This is what the update TS linting told me to do. 🤷 This is pointing to a Class constructor, so strings should _mostly_ be good, but I could see some having symbols, etc. Not sure how that would interact here. 

Do you have thought on a better path?

Reply:
🤷fair enough,  i don't think the current definition should cause any issues. i don't think that the equals statement ever kicks in our current usage, since we're always passing in a T (if you want it to add restrictions  around what T can be, you should use `T extends foo` instead of `T = bar`, or event `T extends foo = bar` to get both).

if we were to use `Constructor<T extends Record<string, unknown>>`, then we could hit typing issues when we try to use symbols.

Reply:
That's super useful. I don't think we'd ever do this here, so you're right that we'll likely be OK.